wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/UserManager.java

```diff
@@ -10,158 +10,204 @@
-    private final UserProfileDao profileDao;
-    private final PreferenceDao  preferenceDao;
-
+    private final UserDao            userDao;
+    private final ProfileManager     profileManager;
+    private final PreferencesManager preferencesManager;
 
     @Inject
-    public UserManager(UserDao userDao,
-                       UserProfileDao profileDao,
-                       PreferenceDao preferenceDao) {
+    public UserManager(UserDao userDao, ProfileManager profileManager, PreferencesManager preferencesManager) {
         this.userDao = userDao;
-        this.profileDao = profileDao;
-        this.preferenceDao = preferenceDao;
+        this.profileManager = profileManager;
+        this.preferencesManager = preferencesManager;
     }
 
-
     /**
-     * Creates new user.
+     * Creates new user and his profile.
      *
-     * @param user
-     *         POJO representation of user entity
+     * @param newUser
+     *         created user
+     * @throws NullPointerException
+     *         when {@code newUser} is null
      * @throws ConflictException
-     *         when given user cannot be created
+     *         when user with such name/email/alias already exists
      * @throws ServerException
      *         when any other error occurs
      */
-    public void create(User user, boolean isTemporary) throws ConflictException, ServerException {
-        user.withId(generate("user", ID_LENGTH))
-            .withPassword(firstNonNull(user.getPassword(), generate("", PASSWORD_LENGTH)));
-        userDao.create(user);
-
-        profileDao.create(new Profile(user.getId()));
-
-        final Map<String, String> preferences = new HashMap<>(4);
-        preferences.put("temporary", Boolean.toString(isTemporary));
-        preferences.put("codenvy:created", Long.toString(System.currentTimeMillis()));
+    public User create(User newUser, boolean isTemporary) throws ConflictException, ServerException {
+        requireNonNull(newUser, "Required non-null user");
+        final UserImpl user = new UserImpl(generate("user", ID_LENGTH),
+                                           newUser.getEmail(),
+                                           newUser.getName(),
+                                           firstNonNull(newUser.getPassword(), generate("", PASSWORD_LENGTH)),
+                                           newUser.getAliases());
         try {
-            preferenceDao.setPreferences(user.getId(), preferences);
-        } catch (NotFoundException e) {
-            LOG.warn(format("Cannot set creation time preferences for user %s.", user.getId()), e);
+            userDao.create(user);
+            profileManager.create(new ProfileImpl(user.getId(), newUser.getEmail()));
+            preferencesManager.save(user.getId(), ImmutableMap.of("temporary", Boolean.toString(isTemporary),
+                                                                  "codenvy:created", Long.toString(currentTimeMillis())));
```

Comment:
@gazarenkov i have nothing to say about the `temporary` preference as far as i know it was used to indicate whether user is temporary, because we don't provide such kind of information in user model, but again i am not sure about it. I think @skabashnyuk should know, + about the 1st statement, how will on-demand creation work for `codenvy:created`?